### PR TITLE
Markdown paste

### PR DESCRIPTION
**Pull request by kof**
## Description

Almost complete markdown support over "paste". 

Missing from parser: strikethrough and table.
Missing components (will add in the next pr): blockquote, list, listItem, separator, code, inlineCode

## Steps for reproduction

Paste this:

```
# Hi this is title 1

This is  _italic_  __bold__ and [link](https://webstudio.is)

## Hi this is title 2

This is  _italic_  __bold__ and [link](https://webstudio.is)

![an image](https://user-images.githubusercontent.com/52824/219961575-cfbc96d6-91c3-4c48-a81e-9e166c001eec.png)

```

https://user-images.githubusercontent.com/52824/219961672-17e07d0c-8929-40c2-a94a-cdfd33a85d63.mp4

## Code Review

- [ ] hi @TrySound , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/shared/copy-paste/plugin-markdown.test.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    Prop.parse(prop);
+  }
+  return { instances, props };
+};
+
+const options = { generateId: () => "123" };
+
+describe("Plugin Markdown", () => {
+  test("paragraph", () => {
+    expect(parseInstanceData(parse("xyz", options))).toMatchSnapshot();
```
Comment (TrySound) [inline apps/builder/app/shared/copy-paste/plugin-markdown.test.ts:31]: Uncolocated snapshots not very readable

Comment (kof) [reply apps/builder/app/shared/copy-paste/plugin-markdown.test.ts:31]: I started with inline snapshots but they are quite big, the entire test becomes to big, was hard to navigate too, inline is better when snapshots are small

Comment (kof) [reply apps/builder/app/shared/copy-paste/plugin-markdown.test.ts]: moved them anyway

File: apps/builder/app/shared/copy-paste/plugin-markdown.ts [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  });
+};
+
+const toInstancesData = (
+  ast: { children: Root["children"] },
+  options: Options = {},
+  props: Array<Prop> = []
+): { instances: Instance["children"]; props: Array<Prop> } => {
+  const { generateId = nanoid } = options;
+  const instances: Instance["children"] = [];
```
Comment (TrySound) [inline apps/builder/app/shared/copy-paste/plugin-markdown.ts:69]: Not instances, can be text child

File: apps/builder/app/shared/copy-paste/plugin-markdown.ts [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    node,
+    props,
+  }: {
+    component: Instance["component"];
+    node: Root["children"][number];
+    props: Array<Prop>;
+  },
+  options: Options
+) => {
+  const { generateId = nanoid } = options;
```
Comment (TrySound) [inline apps/builder/app/shared/copy-paste/plugin-markdown.ts:54]: Why do we have this generateId?

Comment (TrySound) [reply apps/builder/app/shared/copy-paste/plugin-markdown.ts:54]: Ah, got it. For snapshots. Usually solve this with `.toEqual({id: expect.any(string)})`

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/shared/copy-paste/plugin-markdown.ts (hunk 3 above)
Comment (kof) [reply apps/builder/app/shared/copy-paste/plugin-markdown.ts:54]: but you can't use that in snapshots, can you?

File: apps/builder/app/shared/copy-paste/plugin-markdown.ts (hunk 2 above)
Comment (kof) [reply apps/builder/app/shared/copy-paste/plugin-markdown.ts]: true, fixed naming

**Pull request merged**